tests/scripts/validate_modified_files.sh

```diff
@@ -9,26 +9,31 @@ MOD_ERR="changes found by mod.check. You may need to run make clean"
 CRD_ERR="changes found by 'make crds'. please run 'make crds' locally and update your PR"
 BUILD_ERR="changes found by make build', please commit your go.sum or other changed files"
 HELM_ERR="changes found by 'make gen-rbac'. please run 'make gen-rbac' locally and update your PR"
+DOCS_ERR="changes found by 'make docs'. please run 'make docs'locally and update your PR"
+
 
 #############
 # FUNCTIONS #
 #############
 function validate(){
-  git=$(git status --porcelain)
-  for file in $git; do
-    if [ -n "$file" ]; then
-      echo "$1"
-      echo "$git"
-      git diff
+  ERR=$1
+  git diff  --exit-code
+  ret=$?
+  if [ "$ret" != "0" ]; then
+    echo "modidied files detected."
+    git status --porcelain
```

Comment:
Earlier,  the function  would only  run `git status --porcelain` to get the changed files and then print those files.  Now it runs `git diff --exit-code` as well as `git status --porcelain`.  So we are making an extra `git` call in the refracted function.

Reply:
@sp98  wrote:

> Earlier, the function would only run `git status --porcelain` to get the changed files and then print those files. Now it runs `git diff --exit-code` as well as `git status --porcelain`. So we are making an extra `git` call in the refracted function.




This change removes the original mechanism with `git status --porcelain` because I found it complicated and non-obvious.

The new variant uses `git diff --exit-code` to determine if there are changed files. The original code used `git status --porcelain`to detect this.

The new code still uses `git status --porcelain` but to simply print the list of modified files at the end. The original code used this git call, captured its output and printed the list of files by iterating over the lines of output and printing each line in turn.

So yes, there are now two git invocations instead of one:

- one to get the information if modified files exist
- one to print the list of modified files.

I am not sure if `git` invocations are so expensive that we should avoid them, but I think that the new version of the function is a lot simpler then the original one which was, IMHO, unnecessarily complicated.


by the way and for what it's worth, the change you are commenting on is part of the earlier PR #14672  and included in this one only because I based it on the other one. Tis commit shoud vanish from  this PR if the other gets merged.